src/database/models/image_item.rs

```diff
@@ -0,0 +1,287 @@
+
+        Ok(())
+    }
+
+    pub async fn remove(
+        id: ImageId,
+        transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+        redis: &deadpool_redis::Pool,
+    ) -> Result<Option<()>, DatabaseError> {
+        let image = Self::get_id(id, &mut *transaction, redis).await?;
+
+        if let Some(image) = image {
+            sqlx::query!(
+                "
+                DELETE FROM uploaded_images
+                WHERE id = $1
+                ",
+                id as ImageId,
+            )
+            .execute(&mut *transaction)
+            .await?;
+
+            Image::clear_cache(image.id, image.url, redis).await?;
+
+            Ok(Some(()))
+        } else {
+            Ok(None)
+        }
+    }
+
+    pub async fn get_many_contexted<'a, E>(
+        context: ImageContext,
+        exec: E,
+    ) -> Result<Vec<Image>, sqlx::Error>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        use futures::stream::TryStreamExt;
+
+        sqlx::query!(
+            "
+            SELECT i.id, i.url, i.size, i.created, i.owner_id, i.context, i.context_id
+            FROM uploaded_images i
+            WHERE i.context = $1
+            AND i.context_id = $2
+            GROUP BY i.id
+            ",
+            context.context_as_str(),
+            context.inner_id().map(|x| x as i64)
+        )
+        .fetch_many(exec)
+        .try_filter_map(|e| async {
+            Ok(e.right().map(|row| {
+                let id = ImageId(row.id);
+
+                Image {
+                    id,
+                    url: row.url,
+                    size: row.size as u64,
+                    created: row.created,
+                    owner_id: UserId(row.owner_id),
+                    context: ImageContext::from_str(&row.context, row.context_id.map(|x| x as u64)),
+                }
+            }))
+        })
+        .try_collect::<Vec<Image>>()
+        .await
+    }
+
+    pub async fn get<'a, 'b, E>(
+        string: &str,
+        executor: E,
+        redis: &deadpool_redis::Pool,
+    ) -> Result<Option<Image>, DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        Image::get_many(&[string], executor, redis)
+            .await
+            .map(|x| x.into_iter().next())
+    }
+
+    pub async fn get_id<'a, 'b, E>(
+        id: ImageId,
+        executor: E,
+        redis: &deadpool_redis::Pool,
+    ) -> Result<Option<Image>, DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
+        Image::get_many(&[crate::models::ids::ImageId::from(id)], executor, redis)
+            .await
+            .map(|x| x.into_iter().next())
+    }
+
+    pub async fn get_many<'a, E, T: ToString>(
+        image_strings: &[T],
+        exec: E,
+        redis: &deadpool_redis::Pool,
+    ) -> Result<Vec<Image>, DatabaseError>
+    where
+        E: sqlx::Executor<'a, Database = sqlx::Postgres>,
+    {
```

Comment:
don't see when this would be used. when would you specifically need to get an image from it's URL?